packages/kit/src/runtime/client/client.js

```diff
@@ -1810,9 +1810,19 @@ export function create_client(app, target) {
 					// attempt to scroll to that element and avoid any history changes.
 					// Otherwise, this can cause Firefox to incorrectly assign a null
 					// history state value without any signal that we can detect.
-					if (current.url.hash === url.hash) {
+					const [, current_hash] = current.url.href.split('#');
```

Comment:
This change is needed because `current.url.hash` is '' if the url is https://localhost:5173/# or if the url is https://localhost:5173. We need to differentiate https://localhost:5173/# from https://localhost:5173

Reply:
it'd be nice to add a test showing that `'#'` now works differently than `''` to prevent anyone from accidentally breaking it in the future.

Reply:
That's a good idea. Here's a way to test that this patch fixes the issue.

```
<a href="#">Go to Top of Page</a>
```

```
test('clicking a link to # from / takes you to the top of the page', async ({
	page,
	clicknav,
	baseURL
}) => {
	await page.goto('/');
	await clicknav('a[href="#"]');
	expect(new URL(page.url()).href).toBe(`${baseURL}/#`);
});
```

If you're on '/' and you click on a link to '#' it should take you to '/#'. Right now it keeps you on '/'.

I would like to add a test that highlights the difference between '#' and '', but in both cases you end up on the same page.

**Behavior After Fix**
Start on https://localhost:5173/ --> Click on a link to '#' --> End up on https://localhost:5173/#
Start on https://localhost:5173/# --> Click on a link to '#' --> End up on https://localhost:5173/#

I think this test effectively shows the difference between the 2 since if '' and '#' are treated as the same then the test fails.

**Behavior Before Fix**
Start on https://localhost:5173/ --> Click on a link to '#' --> End up on https://localhost:5173/

If you're okay with that I'll update the pull request.